src/components/Header/HeaderMenuMobile/HeaderMenuMobile.tsx

```diff
@@ -0,0 +1,129 @@
+import { useTranslations } from 'next-intl';
+import React from 'react';
+import { MenuImage } from './MenuImage';
+import s from './HeaderMenuMobile.module.scss';
+import { menuOptions } from './mobileMenuData';
+import Image from 'next/image';
+import { usePostLogoutMutation } from '@/api/auth.api';
+import { setAccessToken } from '@/accessToken';
+import { useRouter } from 'next/navigation';
+import { toast } from 'react-toastify';
+import { Modal } from '@/components/Modals/Modal';
+import { TransparentBtn } from '@/components/Buttons/TransparentBtn';
+import { PrimaryBtn } from '@/components/Buttons/PrimaryBtn';
+import { useAppSelector } from '@/redux/hooks/useSelect';
+
+type TProps = {
+  language: string;
+};
+
+export const HeaderMenuMobile = ({ language }: TProps) => {
+  const t = useTranslations('Navigation');
+  const userEmail = useAppSelector((state) => state.auth.user?.email);
+  const [logout, { isLoading }] = usePostLogoutMutation();
+  const [showLogoutModal, setShowLogoutModal] = React.useState(false);
+  const [modal, setModal] = React.useState(false);
+  const tf = useTranslations('Header');
+  const router = useRouter();
+  const onBlurHandler = (e: React.FocusEvent<HTMLTextAreaElement, Element>) => {
+    if (e.relatedTarget?.id === 'mobileMenu') {
+      return;
+    }
+    setModal(false);
+  };
+
+  const keyDownHandler = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
```

Comment:
you can import KeyboardEvent, useState and MouseEvent without React

Reply:
thanks, fix it in next task (refactor)